terraform/aws/modules/hana_node/outputs.tf

```diff
@@ -29,3 +29,13 @@ output "subnets_by_az" {
     s.availability_zone => s.id
   }
 }
+
+output "volume_id_to_device_name" {
+  value = merge([
+    for inst in aws_instance.hana :
+    { for bd in inst.ebs_block_device :
+      bd.volume_id => bd.device_name
+    }
+  ]...)
```

Comment:
what `merge` and the 3 `...` does?

Reply:
3 dots is unpacking the list because the code before creates a list like [key-value1, key-value2 etc], and `merge` gives an error if you try to merge a single list (`like merge([key-value1, key-value2])`)

merge is used to create a single dict out of all the key-value pairs, meaning turn it into a single 
{
key1: value1,
key2: value2
...
}

Reply:
This is me trying to create a valid json as final output